[WGSL] Update keywords and reserved words

tadeuzagallo opened the pull request:
#### e6190f0648b1857508e3b594410cd5d00aadfbba
<pre>
[WGSL] Update keywords and reserved words
<a href="https://bugs.webkit.org/show_bug.cgi?id=260850">https://bugs.webkit.org/show_bug.cgi?id=260850</a>
rdar://114617007

Reviewed by Dan Glastonbury.

Update the set of keywords and reserved words as defined by the spec in
<a href="https://www.w3.org/TR/WGSL/#keyword-summary">https://www.w3.org/TR/WGSL/#keyword-summary</a> and <a href="https://www.w3.org/TR/WGSL/#reserved-words">https://www.w3.org/TR/WGSL/#reserved-words</a>
respectively. This removes a few keywords that we had that are not actually keywords
and adds a bunch of reserved words. For now, we keep the special treatment for the
`array` identifier, as more work is required to treat array types as regular parameterized
types.

* Source/WebGPU/WGSL/Lexer.cpp:
(WGSL::Lexer&lt;T&gt;::lex):
* Source/WebGPU/WGSL/Parser.cpp:
(WGSL::Parser&lt;Lexer&gt;::parseTypeName):
(WGSL::Parser&lt;Lexer&gt;::parseArrayType):
(WGSL::Parser&lt;Lexer&gt;::parseStorageClass):
(WGSL::Parser&lt;Lexer&gt;::parseAccessMode):
(WGSL::Parser&lt;Lexer&gt;::parsePrimaryExpression):
* Source/WebGPU/WGSL/Token.cpp:
(WGSL::toString):
* Source/WebGPU/WGSL/Token.h:

Canonical link: <a href="https://commits.webkit.org/267622@main">https://commits.webkit.org/267622@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d4b5b3777f46b72e4e66cfeee7eb391139baee2b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17161 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17486 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17980 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18947 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16059 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/17353 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20760 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17628 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18255 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17364 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17709 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14899 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19764 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14949 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15590 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22276 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15948 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15757 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20094 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16347 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13867 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15501 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4106 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19868 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16178 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg reviewed (changes requested):
LexerTests need updating

djg reviewed (approved):
:shipit:

tadeuzagallo committed:
[WGSL] Update keywords and reserved words
https://bugs.webkit.org/show_bug.cgi?id=260850
rdar://114617007

Reviewed by Dan Glastonbury.

Update the set of keywords and reserved words as defined by the spec in
https://www.w3.org/TR/WGSL/#keyword-summary and https://www.w3.org/TR/WGSL/#reserved-words
respectively. This removes a few keywords that we had that are not actually keywords
and adds a bunch of reserved words. For now, we keep the special treatment for the
`array` identifier, as more work is required to treat array types as regular parameterized
types.

* Source/WebGPU/WGSL/Lexer.cpp:
(WGSL::Lexer<T>::lex):
* Source/WebGPU/WGSL/Parser.cpp:
(WGSL::Parser<Lexer>::parseTypeName):
(WGSL::Parser<Lexer>::parseArrayType):
(WGSL::Parser<Lexer>::parseStorageClass):
(WGSL::Parser<Lexer>::parseAccessMode):
(WGSL::Parser<Lexer>::parsePrimaryExpression):
* Source/WebGPU/WGSL/Token.cpp:
(WGSL::toString):
* Source/WebGPU/WGSL/Token.h:

Canonical link: https://commits.webkit.org/267622@main
